src/models/backbones/utils/helpers.py

```diff
@@ -0,0 +1,228 @@
+        return state_dict
+    else:
+        _logger.error(f"No checkpoint found at '{checkpoint_path}'")
+        raise FileNotFoundError()
+
+
+def adapt_input_conv(in_chans: int, conv_weight):
+    conv_type = conv_weight.dtype
+    conv_weight = conv_weight.float()  # Some weights are in torch.half, ensure it's float for sum on CPU
+    O, I, J, K = conv_weight.shape
+    if in_chans == 1:
+        if I > 3:
+            assert conv_weight.shape[1] % 3 == 0
+            # For models with space2depth stems
+            conv_weight = conv_weight.reshape(O, I // 3, 3, J, K)
+            conv_weight = conv_weight.sum(dim=2, keepdim=False)
+        else:
+            conv_weight = conv_weight.sum(dim=1, keepdim=True)
+    elif in_chans != 3:
+        if I != 3:
+            raise NotImplementedError('Weight format not supported by conversion.')
+        else:
+            # NOTE this strategy should be better than random init, but there could be other combinations of
+            # the original RGB input layer weights that'd work better for specific cases.
+            repeat = int(math.ceil(in_chans / 3))
+            conv_weight = conv_weight.repeat(1, repeat, 1, 1)[:, :in_chans, :, :]
+            conv_weight *= (3 / float(in_chans))
+    conv_weight = conv_weight.to(conv_type)
+    return conv_weight
+
+
+def load_pretrained(model: nn.Module,
+                    default_cfg: Optional[Dict] = None,
+                    in_chans: int = 3,
+                    filter_fn: Optional[Callable] = None,
+                    strict: bool = False,
+                    progress: bool = False) -> None:
+    """ Load pretrained checkpoint
+
+    Args:
+        model: PyTorch model module
+        default_cfg: default configuration for pretrained weights / target dataset
+        in_chans: in_chans for model
+        filter_fn: state_dict filter fn for load (takes state_dict, model as args)
+        strict: strict load of checkpoint
+        progress: enable progress bar for weight download
+
+    """
+    default_cfg = default_cfg or getattr(model, 'default_cfg', None) or {}
+    pretrained_url = default_cfg.get('url', None)
+    if not pretrained_url:
+        _logger.warning("No pretrained weights exist for this model. Using random initialization.")
+        return
+
+    _logger.info(f'Loading pretrained weights from url ({pretrained_url})')
+    state_dict = load_state_dict_from_url(pretrained_url, progress=progress, map_location='cpu')
+    if 'state_dict' in state_dict:
+        state_dict = state_dict['state_dict']
+
+    if filter_fn is not None:
+        # for backwards compat with filter fn that take one arg, try one first, the two
+        try:
+            state_dict = filter_fn(state_dict)
+        except TypeError:
+            state_dict = filter_fn(state_dict, model)
+
+    input_convs = default_cfg.get('first_conv', None)
+    if input_convs is not None and in_chans != 3:
```

Comment:
It's good feature to load weights from rgb Model to grayscale Model. It must be save.